statsmodels/genmod/generalized_estimating_equations.py

```diff
@@ -1122,26 +1127,42 @@ def fit(self, maxiter=60, ctol=1e-6, start_params=None,
                           ConvergenceWarning)
             return None
 
-        bcov, ncov, bc_cov, _ = self._covmat()
+        bcov, ncov, _ = self._covmat()
         if bcov is None:
             warnings.warn("Estimated covariance structure for GEE "
                           "estimates is singular", ConvergenceWarning)
             return None
+        bc_cov = None
+        if cov_type == "bias_reduced":
+            bc_cov = self._bc_covmat(ncov)
 
         if self.constraint is not None:
-            mean_params, bcov = self._handle_constraint(mean_params, bcov)
+            if cov_type == "robust":
+                cov = bcov
+            elif cov_type == "naive":
+                cov = ncov
+            elif cov_type == "bias_reduced":
+                cov = bc_cov
+
+            mean_params, cov = self._handle_constraint(mean_params, cov)
             if mean_params is None:
                 warnings.warn("Unable to estimate constrained GEE "
                               "parameters.", ConvergenceWarning)
                 return None
+            if cov_type == "robust":
+                bcov = cov
+            elif cov_type == "naive":
+                ncov = cov
+            elif cov_type == "bias_reduced":
+                bc_cov = cov
```

Comment:
I will look at this again and ping you before we merge.



On Mon, May 30, 2016 at 12:20 AM, Josef Perktold <notifications@github.com>
wrote:

> In statsmodels/genmod/generalized_estimating_equations.py
> <https://github.com/statsmodels/statsmodels/pull/2833#discussion_r65022205>
> :
>
> >              if mean_params is None:
> >                  warnings.warn("Unable to estimate constrained GEE "
> >                                "parameters.", ConvergenceWarning)
> >                  return None
> > +            if cov_type == "robust":
> > +                bcov = cov
> > +            elif cov_type == "naive":
> > +                ncov = cov
> > +            elif cov_type == "bias_reduced":
> > +                bc_cov = cov
>
> @kshedden <https://github.com/kshedden> Do you still want to look at this
> or do we merge as is?
> (I'm trying to work my way through some of the PR backlog.)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/statsmodels/statsmodels/pull/2833/files/45086b1d4ddcc6b10d336f4efaa8db4b95cb7ec0#r65022205>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACiww0--eJ7NduOE6UOKw0NVnmcSe-p-ks5qGmWFgaJpZM4HeiHE>
> .
>